### DISCUSSION TITLE
Remove/Skip all pending CoachMarks

### DISCUSSION THREAD

**Discussion by hassan-devcrew** [ephread/Instructions#294] [Q&A]
Basically I want to skip or removed all instructions views which is not displayed yet because another view controller is presented on it before starting the coach & its showing on wrong view controller. So, I want to remove / dismiss all coach marks in this case.

Thanks.

**Comment by ephread** [comment]
To end the walkthrough, you can use `CoachMarksController.stop()`. This method will also reset the internal state of `CoachMarksController`, so you can safely update `dataSource` if you want to start the flow again with another set of coach marks.
